x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/ExtractedFieldsDetector.java

```diff
@@ -35,24 +38,24 @@
 
 public class ExtractedFieldsDetector {
 
+    private static final Logger LOGGER = LogManager.getLogger(ExtractedFieldsDetector.class);
+
     /**
      * Fields to ignore. These are mostly internal meta fields.
      */
     private static final List<String> IGNORE_FIELDS = Arrays.asList("_id", "_field_names", "_index", "_parent", "_routing", "_seq_no",
-        "_source", "_type", "_uid", "_version", "_feature", "_ignored");
+        "_source", "_type", "_uid", "_version", "_feature", "_ignored", DataFrameAnalyticsIndex.ID_COPY);
 
-    /**
-     * The types supported by data frames
-     */
-    private static final Set<String> COMPATIBLE_FIELD_TYPES;
+    public static final Set<String> CATEGORICAL_TYPES = Collections.unmodifiableSet(new HashSet<>(Arrays.asList("text", "keyword", "ip")));
```

Comment:
This could be rewritten to shorter and more readable:
... CATEGORICAL_TYPES = Set.of("text", ...);
at the cost of having a discrepancy between master and 7.x.

I prefer "Set.of" in master but leaving the choice up to you.

Reply:
Me too but I can't be bothered dealing with the backport :-) I think we'll have to wait a bit more (until master becomes 9) and then we can use Set.of everywhere.